client/src/test/java/com/tigrisdata/db/client/StandardTigrisCollectionTest.java

```diff
@@ -109,6 +113,43 @@ public void testReadOneEmpty() throws TigrisException {
     Assert.assertFalse(result.isPresent());
   }
 
+  @Test
+  public void testSearch() throws TigrisException {
+    TigrisClient client = TestUtils.getTestClient(SERVER_NAME, grpcCleanup);
+    TigrisDatabase db1 = client.getDatabase("db1");
+    TigrisCollection<DB1_C1> collection = db1.getCollection(DB1_C1.class);
+    SearchRequest searchRequest =
+        SearchRequest.newBuilder(QueryString.newBuilder("my search string").build())
+            .facetQuery(FacetFields.newBuilder().withField("name").build())
+            .readFields(ReadFields.newBuilder().includeField("other_field").build())
+            .filter(
+                Filters.and(
+                    Filters.eq("first_key", "first_value"), Filters.eq("some_key", "some value")))
```

Comment:
It will be nice to simplify this further like this

```
SearchRequest.newBuilder("my search string")
  .withFacetField("name")
  .withReadFields("other_field", "some_other_field") // withoutReadFields to construct exclusion
  .filter(....);
```

and

```
SearchRequest.newBuilder("my search string")
  .withFields(someMap) // and similarly withFieldOption("fieldName", facetFieldOption);
  .readFields("other_field", "some_other_field")
  .filter(....);
```

Reply:
WIll do this improvement in next iteration